nfv_sriov.md

```diff
@@ -0,0 +1,253 @@
+# NFV SRIOV
+
+This document describes how to deploy EDPM nodes with SRIOV.
+
+## Procedure Overview
+
+1. Configure kernel args and tuned parameters
+2. Configure SRIOV parameters
+3. Configure the networks with SRIOV interfaces of the EDPM nodes
+2. Configure Nova and Neutron parameters.
+
+In order to complete the above procedure, the `services` list of the
+`OpenStackDataPlaneNodeSet` CR needs to be edited.
+
+## OpenStackDataPlaneNodeSet services list
+
+EDPM nodes can be configured by creating an
+`OpenStackDataPlaneNodeSet` CR which the
+[dataplane-operator](https://openstack-k8s-operators.github.io/dataplane-operator)
+will reconcile to create OpenStackDataPlaneService resources
+when an `OpenStackDataPlaneDeployment` CR is created.
+These types of CRs have a `services` list like the following:
+
+```yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneNodeSet
+spec:
+  ...
+  services:
+    - bootstrap
+    - download-cache
+    - configure-network
+    - validate-network
+    - install-os
+    - configure-os
+    - ssh-known-hosts
+    - run-os
+    - reboot-os
+    - install-certs
+    - libvirt
+    - ovn
+    - neutron-ovn
+    - nova-custom-sriov
+    - neutron-sriov
+    - neutron-metadata
+```
+Only the services which are on the list will be configured.
+
+## Configure the EDPM ansible variables of the EDPM nodes
+
+Following are the list of EDPM ansible variables which need to be
+provided for deploying with SRIOV support.
+```
+edpm_kernel_args: Configures kernel boot parameters
+
+edpm_tuned_profile: tuned profile configuration
+
+edpm_tuned_isolated_cores: List of isolated cores
+
+edpm_nova_libvirt_qemu_group: Configures qemu group for nova libvirt
+```
+
+Example:
+```
+edpm_kernel_args: "default_hugepagesz=1GB hugepagesz=1G hugepages=64 iommu=pt intel_iommu=on tsx=off isolcpus=2-11,14-23"
+edpm_tuned_profile: "cpu-partitioning"
+edpm_tuned_isolated_cores: "2-11,14-23"
+edpm_nova_libvirt_qemu_group: "hugetlbfs"
+```
+
+edpm_network_config_template:
+SRIOV enabled network interfaces should be specified in the network config templates to configure SRIOV on the EDPM node.
+
+```
+edpm_network_config_template
+  -
+    type: sriov_pf
+    name: nic3
+    numvfs: 10
+    use_dhcp: false
+    promisc: true
+```
+
+This example also assumes that the EDPM nodes:
```

Comment:
Please turn it into note and move closer to the headline. Ideally between list of vars and example itself.

Reply:
done